packages/slice-machine/src/legacy/components/Card/index.tsx

```diff
@@ -3,8 +3,13 @@ import { Card as ThemeCard, ThemeUIStyleObject } from "theme-ui";
 
 import { CardBox, CardBoxProps } from "./CardBox";
 
+// passing Header as a function caused a bug in EditModal
+// where the function was re-defined after each state update
+// which led to a bug where context was lost and a modal button was not working as expected
+// TODO: tech debt issue DT-2384
```

Comment:
Let's remove this comment from here, as technically there's nothing wrong with this API. The problem is in the caller side (where we also left a comment), where they were defining the component in render.